### DISCUSSION TITLE
Facing db connection issue with nestjs on serverless

### DISCUSSION THREAD

**Discussion by nikhilzoom** [prisma/prisma#24831] [Q&A]
### Question

I have my application on nestjs and prisma used for DB operations. Using serverless deployment.
When performing load testing getting 
`
{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"Error: connect EPIPE 172.31.93.15:3306","reason":{"errorType":"Error","errorMessage":"connect EPIPE 172.31.93.15:3306","code":"EPIPE","errno":-32,"syscall":"connect","address":"172.31.93.15","port":3306,"fatal":true,"stack":["Error: connect EPIPE 172.31.93.15:3306"," at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16)"]},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: Error: connect EPIPE 172.31.93.15:3306"," at process.<anonymous> (file:///var/runtime/index.mjs:1276:17)"," at process.emit (node:events:517:28)"," at process.emit (node:domain:489:12)"," at emit (node:internal/process/promises:149:20)"," at processPromiseRejections (node:internal/process/promises:283:27)"," at process.processTicksAndRejections (node:internal/process/task_queues:96:32)"]}
`
Any idea why I am getting this, and how i can resolve this

I am using following
Prisma 5.17
Node 18.20.3
NestJs 8.0.0
DB: MySQL RDS(AWS) with RDS Proxy

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @nikhilzoom 👋 

Thank you for raising this question.

The error you're encountering, `connect EPIPE`, typically indicates that the connection to your database was broken. Serverless environments can cause issues with database connection pooling. Prisma's query engine maintains a connection pool, which might lose its effectiveness in serverless environments. You can read more about this in the [documentation on connection management](https://www.prisma.io/docs/guides/performance-and-optimization/connection-management#serverless-environments-faas). 

Can you take a look at the logs of your MySQL database for any signs of connection issues or other errors you can share to help us investigate this further?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
